py/desispec/io/photo.py

```diff
@@ -46,93 +53,109 @@ def gather_targetdirs(tileid, fiberassign_dir=None):
+        if targetclass in fahdr:
+            alltargetdirs += [fahdr[targetclass]]
+            for num in (np.arange(99)+2).astype(str):
+                moretarg = targetclass+num
+                if moretarg in fahdr and fahdr[moretarg].strip() != '-':
+                    alltargetdirs += [fahdr[moretarg]]
+                else:
+                    break
+
+    # Checks taken from fiberassign.io.fba_patch_io.get_static_desitarget_fns
+
+    # Special case 80615 and 80736.
+    if tileid == 80615:
+        alltargetdirs += [os.path.join(desi_root, 'target', 'catalogs', 'gaiadr2', '0.47.0', 'targets', 'cmx', 'resolve', 'supp')]
+
+    # Fix paths (from AR, fiberassign.io.fba_patch_io.get_static_desitarget_fns)
+    targetdirs = []    
+    for ii, fn in enumerate(alltargetdirs):
+        fn = fn.replace(
+            "DESIROOT/target/catalogs/mtl/1.1.1/mtl/main/ToO/ToO.ecsv",
+            "{}/mtl/main/ToO/ToO.ecsv".format(os.getenv("DESI_SURVEYOPS")),
+        )
+        fn = fn.replace(
+            "DESIROOT/survey/ops/staging/mtl/main/ToO/ToO.ecsv",
+            "{}/mtl/main/ToO/ToO.ecsv".format(os.getenv("DESI_SURVEYOPS")),
+        )
+        ## AR case where the path is generically defined, but not used (no mtl for sv1)
+        #fn = fn.replace(
+        #    "DESIROOT/survey/ops/surveyops/trunk/mtl/sv1/ToO/ToO.ecsv",
+        #    "-",
+        #)
+        ## AR case where the path is generically defined, but not used (no mtl for sv1)
+        #fn = fn.replace(
+        #    "DESIROOT/survey/ops/surveyops/trunk/mtl/sv2/ToO/ToO.ecsv",
+        #    "-",
+        #)
+        fn = fn.replace(
+            "DESIROOT/target/catalogs/dr9/0.58.0/targets/main/secondary/dark/maintargets-dark-secondary.fits",
+            "DESIROOT/target/catalogs/dr9/0.58.0/targets/main/secondary/dark/targets-dark-secondary.fits",
+        )
+        ## AR case where the path is generically defined, but not used (81096-81099; no secondaries for sv2)
+        #fn = fn.replace(
+        #    "DESIROOT/target/catalogs/dr9/0.53.0/targets/sv2/secondary/dark/sv2targets-dark-secondary.fits",
+        #    "-",
+        #)
+        if 'secondary' in fn:#name == "scnd":
+            txt = os.path.sep.join(fn.split(os.path.sep)[-3:])
+            if txt in ["sv1/secondary/bright", "sv1/secondary/dark"]:
+                prog = fn.split(os.path.sep)[-1]
+                fn = os.path.join(fn, "sv1targets-{}-secondary.fits".format(prog))
+        fn = fn.replace("DESIROOT", os.getenv("DESI_ROOT"))
+        fn = fn.replace("/data/target", os.getenv("DESI_TARGET"))
+        fn = fn.replace(
+            "/data/afternoon_planning/surveyops/trunk", os.getenv("DESI_SURVEYOPS")
+        )
+        fn = fn.replace(
+            "/global/cscratch1/sd/adamyers/gaiadr2/1.3.0.dev5218/targets/main/resolve/backup",
+            "{}/catalogs/gaiadr2/2.2.0/targets/main/resolve/backup".format(
+                os.getenv("DESI_TARGET")
+            ),
+        )
```

Comment:
for special cases like this, please add a comment for our future selves along the lines of "experimental tile 82401 for developing the backup program used gaiadr2/1.3.0.dev5218 inputs that no longer but are equivalent to gaiadr2/2.2.0"